libs/designer-ui/src/lib/settings/settingsection/settingTokenField.tsx

```diff
@@ -315,3 +328,38 @@ const TokenField = ({
       );
   }
 };
+
+export type CustomTokenFieldProps = TokenFieldProps & { editorOptions: ICustomEditorOptions };
+
+export const CustomTokenField = (props: CustomTokenFieldProps) => {
+  invariant(isCustomEditor(props), 'only custom editor can be rendered by CustomTokenField');
+  const { EditorComponent, editor, editorOptions = {} }: ICustomEditorOptions = props.editorOptions;
+
+  const renderDefaultEditor = useRenderDefaultEditor(props);
+
+  const customEditorProps: IEditorProps = {
+    editor,
+    editorOptions,
+    value: props.value,
+    onValueChange: props.onValueChange as IEditorValueSegmentChangeHandler,
+    renderDefaultEditor,
+  };
+  return <EditorComponent {...customEditorProps} />;
+};
+
+function useRenderDefaultEditor(tokenFieldProps: Omit<TokenFieldProps, 'editor' | 'editor' | 'value' | 'onValueChange'>) {
+  return useCallback(
+    (props: IRenderDefaultEditorParams) => {
+      return (
+        <TokenField
+          {...tokenFieldProps}
+          editor={props.editor}
+          editorOptions={props.editorOptions}
+          value={props.value as ValueSegment[]}
+          onValueChange={props.onValueChange as ChangeHandler}
```

Comment:
instead of converting onValueChange to ChangeHandler, we should change it of type ChangeHandler without having to update the viewmodel